src/nwb_conversion_tools/tools/data_transfers.py

```diff
@@ -329,60 +326,52 @@ def dandi_upload(
+        Defaults to False.
     """
-    initial_wd = os.getcwd()
-    dandiset_folder_path = nwb_folder_path.parent if dandiset_folder_path is None else dandiset_folder_path
+    dandiset_folder_path = (
+        Path(mkdtemp(dir=nwb_folder_path.parent)) if dandiset_folder_path is None else dandiset_folder_path
+    )
+    dandiset_path = dandiset_folder_path / dandiset_id
     version = "draft" if version is None else version
-    process_timeouts = dict() if process_timeouts is None else process_timeouts
     assert os.getenv("DANDI_API_KEY"), (
         "Unable to find environment variable 'DANDI_API_KEY'. "
         "Please retrieve your token from DANDI and set this environment variable."
     )
+
     url_base = "https://gui-staging.dandiarchive.org" if staging else "https://dandiarchive.org"
     dandiset_url = f"{url_base}/dandiset/{dandiset_id}/{version}"
-
-    os.chdir(nwb_folder_path.parent)
-    validate_return = deploy_process(
-        command=f"dandi validate {nwb_folder_path.name}",
-        catch_output=True,
-        timeout=process_timeouts.get("validate"),
-    )
-    assert re.fullmatch(
-        pattern=r"^Summary: No validation errors among \d+ file\(s\)$", string=validate_return
-    ), "DANDI validation failed!"
-
-    os.chdir(dandiset_folder_path)
-    download_return = deploy_process(
-        command=f"dandi download {dandiset_url} --download dandiset.yaml",
-        catch_output=True,
-        timeout=process_timeouts.get("download"),
-    )
-    assert download_return, "DANDI download failed!"  # output is a bit too dynamic to regex; if it fails it is empty
-
-    os.chdir(dandiset_folder_path / dandiset_id)
-    deploy_process(
-        command=f"dandi organize {nwb_folder_path.absolute()}",  # .absolute() needed if dandiset folder is elsewhere
-        timeout=process_timeouts.get("organize", 120.0),
-    )
-    assert len(list(Path.cwd().iterdir())) > 1, "DANDI organize failed!"
-
-    dandi_upload_command = "dandi upload -i dandi-staging" if staging else "dandi upload"
-    upload_return = deploy_process(
-        command=dandi_upload_command, catch_output=True, timeout=process_timeouts.get("upload")
-    )
-    assert upload_return, "DANDI upload failed!"
-
-    # cleanup - should be confirmed manually, Windows especially can complain
-    os.chdir(initial_wd)  # restore to initial working directory; also to prevent additional process usage permissions
-    try:
-        rmtree(path=nwb_folder_path.parent / dandiset_id)
-        rmtree(path=nwb_folder_path)
-    except PermissionError:
-        warn("Unable to clean up source files and dandiset! Please manually delete them.")
+    download(urls=dandiset_url, output_dir=str(dandiset_folder_path), get_metadata=True, get_assets=False)
+    assert dandiset_path.exists(), "DANDI download failed!"
+
+    organize(paths=str(nwb_folder_path), dandiset_path=str(dandiset_path))
+    organized_nwbfiles = dandiset_path.rglob("*.nwb")
```

Comment:
Do you need `rglob` here, would not a simple `glob` suffice?

Reply:
organization goes into subfolders, e.g.,

```
dandiset_folder / dandiset_id / sub-1 / sub-1_ses-2_ecephys.nwb
dandiset_folder / dandiset_id / sub-2 / sub-2_ses-3_ecephys.nwb
```

Reply:
Thanks, I was curios, what does each sub-folder corresponds to?

Reply:
'sub' is dandi shorthand for 'subject'